src/providers/mdal/qgsmdalprovider.cpp

```diff
@@ -493,13 +493,17 @@ QgsMeshDatasetGroupMetadata QgsMdalProvider::datasetGroupMetadata( int groupInde
     metadata[key] = value;
   }
 
+  QString referenceTimeString( MDAL_G_referenceTime( group ) );
+  QDateTime referenceTime = QDateTime::fromString( referenceTimeString, Qt::ISODate );
```

Comment:
what if referenceTimeString is empty string?

Reply:
If the static method of QDateTime fails to parse the string, it returns invalid QDateTime and we considers that the dataset group don't have valid reference time.